esp-hal/src/rmt.rs

```diff
@@ -1103,497 +1093,415 @@ pub trait RxChannel: private::RxChannelInternal<crate::Blocking> {
+            }
+            3 => {
+                <Channel<Async, 3> as TxChannelInternal<Async>>::unlisten_interrupt(Event::End);
+                <Channel<Async, 3> as TxChannelInternal<Async>>::unlisten_interrupt(Event::Error);
+                <Channel<Async, 3> as RxChannelInternal<Async>>::unlisten_interrupt(Event::End);
+                <Channel<Async, 3> as RxChannelInternal<Async>>::unlisten_interrupt(Event::Error);
+            }
+            #[cfg(esp32)]
+            4 => {
+                <Channel<Async, 4> as TxChannelInternal<Async>>::unlisten_interrupt(Event::End);
+                <Channel<Async, 4> as TxChannelInternal<Async>>::unlisten_interrupt(Event::Error);
+                <Channel<Async, 4> as RxChannelInternal<Async>>::unlisten_interrupt(Event::End);
+                <Channel<Async, 4> as RxChannelInternal<Async>>::unlisten_interrupt(Event::Error);
+            }
+            #[cfg(any(esp32, esp32s3))]
+            5 => {
+                <Channel<Async, 5> as TxChannelInternal<Async>>::unlisten_interrupt(Event::End);
+                <Channel<Async, 5> as TxChannelInternal<Async>>::unlisten_interrupt(Event::Error);
+                <Channel<Async, 5> as RxChannelInternal<Async>>::unlisten_interrupt(Event::End);
+                <Channel<Async, 5> as RxChannelInternal<Async>>::unlisten_interrupt(Event::Error);
+            }
+            #[cfg(any(esp32, esp32s3))]
+            6 => {
+                <Channel<Async, 6> as TxChannelInternal<Async>>::unlisten_interrupt(Event::End);
+                <Channel<Async, 6> as TxChannelInternal<Async>>::unlisten_interrupt(Event::Error);
+                <Channel<Async, 6> as RxChannelInternal<Async>>::unlisten_interrupt(Event::End);
+                <Channel<Async, 6> as RxChannelInternal<Async>>::unlisten_interrupt(Event::Error);
+            }
+            #[cfg(any(esp32, esp32s3))]
+            7 => {
+                <Channel<Async, 7> as TxChannelInternal<Async>>::unlisten_interrupt(Event::End);
+                <Channel<Async, 7> as TxChannelInternal<Async>>::unlisten_interrupt(Event::Error);
+                <Channel<Async, 7> as RxChannelInternal<Async>>::unlisten_interrupt(Event::End);
+                <Channel<Async, 7> as RxChannelInternal<Async>>::unlisten_interrupt(Event::Error);
             }
 
-            WAKER[channel].wake();
+            _ => unreachable!(),
         }
-    }
-}
-
-mod private {
-    use crate::{peripheral::Peripheral, soc::constants};
 
-    pub enum Event {
-        Error,
-        Threshold,
-        End,
+        WAKER[channel].wake();
     }
+}
 
-    pub trait CreateInstance<'d> {
-        fn create(peripheral: impl Peripheral<P = crate::peripherals::RMT> + 'd) -> Self;
-    }
+#[doc(hidden)]
+pub enum Event {
+    Error,
+    Threshold,
+    End,
+}
 
-    pub trait TxChannelInternal<M>
-    where
-        M: crate::Mode,
-    {
-        const CHANNEL: u8;
+#[doc(hidden)]
+pub trait TxChannelInternal<M>
```

Comment:
now a user could implement `TxChannelInternal` (and `TxChannel`)  etc. - when removing the `private` mod we need to seal the internal traits, no?

Reply:
These traits shouldn't even exist in the first place, so I don't think this is a big deal.